borg/key.py

```diff
@@ -396,19 +400,30 @@ class KeyfileKey(KeyfileKeyBase):
     TYPE = 0x00
     FILE_ID = 'BORG_KEY'
 
+    def sanity_check(self, filename):
+        with open(filename, 'r') as fd:
+            line = fd.readline().strip()
+            if line.startswith(self.FILE_ID) and line[len(self.FILE_ID) + 1:] == id:
+                return filename
+            else:
+                return None
```

Comment:
I tried to upload another version, which should work with the for loop (ie it returns None instead of raising KeyfileMismatch)